server/datastore/mysql/software.go

```diff
@@ -211,15 +211,34 @@ func (d *Datastore) hostSoftwareFromHostID(tx *sqlx.Tx, id uint) ([]fleet.Softwa
 		selectFunc = tx.Select
 	}
 	sql := `
-		SELECT * FROM software
-		WHERE id IN
+		SELECT s.*, coalesce(scp.cpe, "") as generated_cpe FROM software s
+		LEFT JOIN software_cpe scp ON (s.id=scp.software_id)
+		WHERE s.id IN
 			(SELECT software_id FROM host_software WHERE host_id = ?)
 	`
 	var result []fleet.Software
 	if err := selectFunc(&result, sql, id); err != nil {
 		return nil, errors.Wrap(err, "load host software")
 	}
-	return result, nil
+
+	var resultWithCVE []fleet.Software
+	for _, software := range result {
+		var vulnerabilities []fleet.SoftwareCVE
+		if software.GenerateCPE != "" {
+			vulnSql := `
+				SELECT scv.cve, CONCAT('https://nvd.nist.gov/vuln/detail/', scv.cve) as details_link 
+				FROM software_cve scv
+				JOIN software_cpe scp ON (scv.cpe_id=scp.id)
+				WHERE scp.cpe = ?
+			`
+			if err := selectFunc(&vulnerabilities, vulnSql, software.GenerateCPE); err != nil {
```

Comment:
Is it necessary to do this in separate queries rather than as part of the SELECT on line 214?

Reply:
From what I could see, the best option is to use GROUP_CONCAT, get a string and parse it. It didn't seem great, but I'm happy to try it if you think it's a worth while approach.

Reply:
Hm, or maybe I can use JSON_ARRAYAGGR and have sqlx parse it for me... I'll try.

Reply:
Yeah, ok, pushed. It feels awkward the way it's done, but it's probably faster that way.